plugins/search-backend/src/service/router.ts

```diff
@@ -154,9 +160,19 @@ export async function createRouter(
         const resultSet = await engine?.query(query, { token });
 
         res.send(filterResultSet(toSearchResults(resultSet)));
-      } catch (err) {
+      } catch (error) {
+        if (error instanceof MissingIndexError) {
+          res
+            .status(400)
+            .send(
```

Comment:
Let's avoid using res.send as it could open up for XSS attacks. Some additional context and alternatives in https://github.com/backstage/backstage/issues/4423